scipy/linalg/flapack_other.pyf.src

```diff
@@ -401,6 +401,116 @@ subroutine <prefix2c>unmqr(side,trans,m,n,k,a,lda,tau,c,ldc,work,lwork,info)
 
 end subroutine <prefix2c>unmqr
 
+subroutine <prefix>geqrt(m,n,nb,a,lda,t,ldt,work,info)
+    ! a,t,info = geqrt(nb,a,[overwrite_a=0])
+    !
+    ! Computes a QR factorization with block size nb of a general matrix A,
+    ! using the compact WY representation for Q. T stores the upper triangular
+    ! block reflectors.
+
+    callstatement (*f2py_func)(&m,&n,&nb,a,&lda,t,&ldt,work,&info)
+    callprotoargument int*,int*,int*,<ctype>*,int*,<ctype>*,int*,<ctype>*,int*
+
+    integer intent(hide),depend(a):: m = shape(a,0)
+    integer intent(hide),depend(a):: n = shape(a,1)
+    integer intent(in),depend(m,n),check(MIN(m,n)>=nb&&nb>=1):: nb
+    <ftype> dimension(m,n),intent(in,out,copy):: a
+    integer intent(hide),depend(a):: lda = MAX(1,shape(a,0))
+    <ftype> dimension(nb,MIN(m,n)),intent(out):: t
+    integer intent(hide),depend(t):: ldt = MAX(1,shape(t,0))
+    <ftype> dimension(nb,n),intent(hide,cache):: work
+    integer intent(out):: info
+
+end subroutine <prefix>geqrt
+
+subroutine <prefix>gemqrt(side,trans,m,n,k,nb,v,ldv,t,ldt,c,ldc,work,info)
+    ! c,info = gemqrt(side,trans,v,t,c,[overwrite_c=0])
+    !
+    ! Multiplies a general matrix C by the orthogonal matrix Q defined by the
+    ! elementary reflectors stored in matrix V and the upper triangular block
+    ! reflectors in matrix T. C may be multiplied by Q, its transpose (for real
+    ! matrices), or its adjoint (for complex matrices) from the left or right.
+
+    callstatement (*f2py_func)(side,trans,&m,&n,&k,&nb,v,&ldv,t,&ldt,c,&ldc,work,&info)
+    callprotoargument char*,char*,int*,int*,int*,int*,<ctype>*,int*,<ctype>*,int*,<ctype>*,int*,<ctype>*,int*
+
+    character intent(in),check(*side=='L'||*side=='R'):: side
```

Comment:
These chars best be optional with default values. I know we are not perfectly consistent but we are trying to fix them as we wrap new functions into these files.

Reply:
Having said that we also need a simple test that gives wrong chars as arguments and we expect them to fail.

Reply:
Great, I'm assuming `trans` should also have the default value `'N'`?

Reply:
Yes indeed. And the side is L

Reply:
Alright, done, with tests, in the latest commit.